File: web/check_ui.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
使用 Playwright 检查前端页面渲染效果
"""

import asyncio
from playwright.async_api import async_playwright

async def check_ui():
    async with async_playwright() as p:
        # 启动浏览器
        browser = await p.chromium.launch(headless=False)
        context = await browser.new_context(
            viewport={'width': 1920, 'height': 1080}
        )
        page = await context.new_page()

        # 访问页面
        print("正在加载页面...")
        await page.goto('http://localhost:8080/')

        # 等待页面加载完成
        await page.wait_for_load_state('networkidle')

        # 等待 Vue 应用挂载
        await asyncio.sleep(3)

        # 截图
        await page.screenshot(path='viewport_1920x1080.png')
        print("已截图: viewport_1920x1080.png")

        # 检查是否有错误
        errors = await page.evaluate('''() => {
            return {
                hasVue: typeof window.Vue !== 'undefined',
                appMounted: document.querySelector('#app') !== null,
                appContent: document.querySelector('#app')?.innerText?.substring(0, 100) || '',
                cssLoaded: document.querySelector('link[rel="stylesheet"]') !== null,
                bodyStyles: window.getComputedStyle(document.body).fontFamily,
                scrollToExist: window.scrollBy !== undefined
            };
        }''')

        print("\n页面状态检查:")
        print(f"  Vue 是否加载: {errors['hasVue']}")
        print(f"  App 是否挂载: {errors['appMounted']}")
        print(f"  App 内容: {errors['appContent']}")
        print(f"  CSS 是否加载: {errors['cssLoaded']}")
        print(f"  Body 字体: {errors['bodyStyles']}")

        # 检查 CSS 文件是否成功加载
        css_status = await page.evaluate('''() => {
            const links = document.querySelectorAll('link[rel="stylesheet"]');
            const styles = [];
            links.forEach(link => {
                styles.push({
                    href: link.href,
                    buster: link.buster
                });
            });
            return styles;
        }''')

        print(f"\nCSS 链接: {css_status}")

        # 等待用户查看
        print("\n等待 5 秒供手动查看...")
        await asyncio.sleep(5)

        await browser.close()

if __name__ == '__main__':
    asyncio.run(check_ui())

File: web/check_ui.js
#!/usr/bin/env node
// -*- coding: utf-8 -*-
/**
 * 使用 Playwright 检查前端页面渲染效果
 */

const { chromium } = require('playwright');

async function checkUI() {
  const browser = await chromium.launch({
    headless: false,
    slowMo: 50
  });

  const context = await browser.newContext({
    viewport: { width: 1920, height: 1080 }
  });

  const page = await context.newPage();

  // 访问页面
  console.log('正在加载页面...');
  await page.goto('http://localhost:8080/', {
    waitUntil: 'networkidle',
    timeout: 60000
  });

  // 等待 Vue 应用挂载
  await page.waitForSelector('#app', { timeout: 10000 });
  await new Promise(r => setTimeout(r, 2000));

  // 截图
  await page.screenshot({
    path: 'viewport_1920x1080.png',
    fullPage: true
  });
  console.log('已截图: viewport_1920x1080.png');

  // 获取页面状态
  const pageState = await page.evaluate(() => {
    const app = document.querySelector('#app');
    return {
      appExists: !!app,
      appChildCount: app?.children?.length || 0,
      bodyClass: document.body.className,
      documentTitle: document.title,
      linkCount: document.querySelectorAll('link').length,
      scriptCount: document.querySelectorAll('script').length,
    };
  });

  console.log('\n页面状态:');
  console.log('  App 元素存在:', pageState.appExists);
  console.log('  App 子元素数量:', pageState.appChildCount);
  console.log('  Body 类名:', pageState.bodyClass);
  console.log('  文档标题:', pageState.documentTitle);
  console.log('  Link 标签数量:', pageState.linkCount);
  console.log('  Script 标签数量:', pageState.scriptCount);

  // 检查 CSS 是否加载
  const cssStatus = await page.evaluate(() => {
    const links = Array.from(document.querySelectorAll('link[rel="stylesheet"]'));
    return links.map(link => ({
      href: link.href,
      success: link.href.indexOf('404') === -1
    }));
  });

  console.log('\nCSS 状态:');
  cssStatus.forEach(css => {
    console.log(`  ${css.href}: ${css.success ? '已加载' : '未找到'}`);
  });

  // 等待用户查看
  console.log('\n等待 10 秒供手动查看...');
  await new Promise(r => setTimeout(r, 10000));

  await browser.close();
}

checkUI().catch(console.error);
